var/ramble/repos/builtin/base_classes/package-manager-base/base_class.py

```diff
@@ -210,6 +215,137 @@ def populate_inventory(
 
         pass
 
+    def define_missing_packages(self, workspace):
+        """Injection of missing packages that are auto-injected by objects
```

Comment:
Is this the same but simpler?
```suggestion
        """Auto-inject missing packages by objects
```

Reply:
Not exactly. Ultimately, we only inject packages that are marked as injectable. We do not arbitrarily inject missing packages from objects though.